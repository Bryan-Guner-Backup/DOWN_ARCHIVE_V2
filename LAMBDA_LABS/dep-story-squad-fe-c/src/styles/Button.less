.game-button {
     background-color: @red ;
     color:@white;
     padding:6px 50px;
     outline: none;
     border: 3px solid @black;
     border-radius: 10px ;
     transform:scale(1);
     transition: transform 0.3s ease-in-out;
     font-weight:500;
     font-size:1.2rem;
     &:hover{
      color:@black;
      transform:scale(1.1)
     }
}

.game-button.dark {
     background-color: @brown ;
     border: 3px solid #000;
     border-radius: 10px ;
     transform:scale(1);
     transition: transform 0.3s ease-in-out;
     font-weight:500;
     font-size:1.2rem;
     &:hover{
      color:@red;
      transform:scale(1.1)
     }
}

