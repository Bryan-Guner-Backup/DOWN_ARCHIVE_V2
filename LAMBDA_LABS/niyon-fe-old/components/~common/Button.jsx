import React from 'react';
import styled from 'styled-components';
import { lighten, darken } from 'polished';
import { PacmanLoader } from 'react-spinners';
import { css } from '@emotion/core';
import { theme } from '../../lib/theme';

// 100% width inside media query will make button responsive for different mobile screen sizes.
// Change the percentage the amount of space button fills on smaller screens.

const B = styled.button`
  height: 40px;
  width: ${({ width }) => width || '150px'};
  background: ${({ themeColor }) => `${themeColor}`};
  border: none;
  color: ${({ theme }) => theme.white};
  border-radius: ${({ theme }) => theme.borderRadius};
  transition: background 400ms;
  cursor: pointer;
  display: flex;
  justify-content: center;
  align-items: center;
  text-transform: uppercase;
  font-size: ${({ theme }) => theme.smallText};

  &:hover {
    background: ${({ themeColor }) => `${darken(0.1, themeColor)}`};
  }

  &:focus {
    outline: none;
  }

  @media (max-width: ${({ theme }) => theme.mobileWidth}) {
    width: ${({ width }) => width || '100%'};
  }
`;

const BRaised = styled.button`
  height: 40px;
  width: ${({ width }) => width || '150px'};
  background: ${({ themeColor }) => `${themeColor}`};
  border: none;
  color: ${({ theme }) => theme.white};
  border-radius: ${({ theme }) => theme.borderRadius};
  box-shadow: ${({ theme }) => theme.boxShadow};
  transition: transform 400ms, background 400ms;
  cursor: pointer;
  display: flex;
  justify-content: center;
  align-items: center;
  text-transform: uppercase;
  font-size: ${({ theme }) => theme.smallText};

  &:hover {
    background: ${({ themeColor }) => `${darken(0.1, themeColor)}`};
    transform: translateY(-2px);
    box-shadow: ${({ theme }) => theme.boxShadow};
  }
`;

const BOutline = styled.button`
  height: 40px;
  width: ${({ width }) => width || '150px'};
  background: none;
  border: ${({ themeColor }) => `1px solid ${themeColor}`};
  color: ${({ themeColor }) => `${themeColor}`};
  border-radius: ${({ theme }) => theme.borderRadius};
  transition: border 400ms, background 400ms, color 400ms;
  cursor: pointer;
  text-align: center;
  display: flex;
  justify-content: center;
  align-items: center;
  text-transform: uppercase;
  font-size: ${({ theme }) => theme.smallText};

  &:hover {
    border: ${({ themeColor }) => `1px solid ${darken(0.1, themeColor)}`};
    color: ${({ themeColor }) => `${darken(0.1, themeColor)}`};
    background: #f9f9f9;
  }
`;

function Button(props) {
  const {
    children,
    outline,
    primary,
    secondary,
    warning,
    danger,
    raised,
    small,
    large,
    loadingB
  } = props;

  // Setting the color
  let color;
  if (primary) {
    color = `${theme.primary}`;
  } else if (secondary) {
    color = `${theme.secondary}`;
  } else if (warning) {
    color = `${theme.warning}`;
  } else if (danger) {
    color = `${theme.danger}`;
  }

  let width;
  if (small) {
    width = '150px';
  } else if (large) {
    width = '250px';
  }

  // Override the css for the react-spinners
  const override = css`
    margin: 0 auto;
    width: 30px;
  `;

  if (outline) {
    return (
      <BOutline type="button" themeColor={color} width={width} {...props}>
        {loadingB ? (
          <PacmanLoader
            loading={true}
            color={theme.white}
            css={override}
            sizeUnit={'px'}
            size={10}
          />
        ) : (
          children
        )}
      </BOutline>
    );
  }
  if (raised) {
    return (
      <BRaised type="button" themeColor={color} width={width} {...props}>
        {loadingB ? (
          <PacmanLoader
            loading={true}
            color={theme.white}
            css={override}
            sizeUnit={'px'}
            size={10}
          />
        ) : (
          children
        )}
      </BRaised>
    );
  }

  return (
    <B type="button" themeColor={color} width={width} {...props}>
      {loadingB ? (
        <PacmanLoader
          loading={true}
          color={theme.white}
          sizeUnit={'px'}
          css={override}
          size={10}
        />
      ) : (
        children
      )}
    </B>
  );
}

export default Button;
