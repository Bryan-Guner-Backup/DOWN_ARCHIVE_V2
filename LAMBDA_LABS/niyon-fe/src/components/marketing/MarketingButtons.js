import React from 'react'
import Button from '@material-ui/core/Button'
import { makeStyles } from '@material-ui/core/styles'
import { Link } from 'react-router-dom'

const useStyles = makeStyles((theme) => ({
  root: { display: 'flex' },
  buttons: {
    margin: '20% auto 0 auto',
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
    paddingTop: '10px'
  },
  button: {
    width: '250px',
    borderRadius: '50px',
    height: '50px',
    margin: '15px 0',
    textTransform: 'capitalize'
  }
}))

function MarketingButtons () {
  const classes = useStyles()
  return (
    <div className="marketing-container" data-test="marketing-container">
      <div className="container" data-test="marketing-second-container">
        <h3 data-test="container-title">Connect with Mentors in your area!</h3>
      </div>
      <div className={classes.buttons} data-test="buttons-container">
        <Link to="/registration">
          <Button
            variant="contained"
            color="primary"
            className={classes.button}
          >
            Sign Up
          </Button>
        </Link>
        <Link to="/login">
          <Button
            variant="contained"
            color="secondary"
            className={classes.button}
          >
            Login
          </Button>
        </Link>
      </div>
    </div>
  )
}

export default MarketingButtons
