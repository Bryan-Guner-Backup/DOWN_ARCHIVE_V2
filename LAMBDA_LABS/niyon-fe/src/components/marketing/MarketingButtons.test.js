import React from 'react'
import toJSON from 'enzyme-to-json'
import { shallow } from 'enzyme'
import Button from '@material-ui/core/Button'
import findByTestAttr from '../../tests/utils'
import MarketingButtons from './MarketingButtons'

const setUp = (props = {}, state = null) => {
  const wrapper = shallow(<MarketingButtons />)
  return wrapper
}

describe('Marketing container testing', () => {
  let component
  beforeEach(() => {
    component = setUp()
  })

  it('should pass snapshot testing correctly', () => {
    expect(toJSON(component)).toMatchSnapshot()
  })

  it('should render Marketing container correctly', () => {
    const mainContainer = findByTestAttr(component, 'marketing-container')
    expect(mainContainer.length).toBe(1)
  })

  it('should render second container div correctly', () => {
    const secondContainer = findByTestAttr(component, 'marketing-second-container')
    expect(secondContainer.length).toBe(1)
  })

  it('should render container title correctly', () => {
    const h3 = findByTestAttr(component, 'container-title')
    expect(h3.length).toBe(1)
    expect(h3.text()).toEqual('Connect with Mentors in your area!')
  })
  it('should render buttons container correctly', () => {
    const buttonsContainer = findByTestAttr(component, 'buttons-container')
    expect(buttonsContainer.length).toBe(1)
  })

  it('should render Sign Up button with no errors', () => {
    const signUpButton = findByTestAttr(component, 'buttons-container').find(Button).at(0)
    expect(signUpButton.exists()).toBe(true)
    expect(signUpButton.text()).toEqual('Sign Up')
  })

  it('should render Login button with no errors', () => {
    const logInButton = findByTestAttr(component, 'buttons-container').find(Button).at(1)
    expect(logInButton.exists()).toBe(true)
    expect(logInButton.text()).toEqual('Login')
  })
})
