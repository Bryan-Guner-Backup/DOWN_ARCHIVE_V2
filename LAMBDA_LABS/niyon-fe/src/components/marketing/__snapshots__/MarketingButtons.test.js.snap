// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Marketing container testing should pass snapshot testing correctly 1`] = `
<div
  className="marketing-container"
  data-test="marketing-container"
>
  <div
    className="container"
    data-test="marketing-second-container"
  >
    <h3
      data-test="container-title"
    >
      Connect with Mentors in your area!
    </h3>
  </div>
  <div
    className="makeStyles-buttons-2"
    data-test="buttons-container"
  >
    <Link
      to="/registration"
    >
      <WithStyles(ForwardRef(Button))
        className="makeStyles-button-3"
        color="primary"
        variant="contained"
      >
        Sign Up
      </WithStyles(ForwardRef(Button))>
    </Link>
    <Link
      to="/login"
    >
      <WithStyles(ForwardRef(Button))
        className="makeStyles-button-3"
        color="secondary"
        variant="contained"
      >
        Login
      </WithStyles(ForwardRef(Button))>
    </Link>
  </div>
</div>
`;
